### PR TITLE
ci: support `naming-convention`

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/addon-editor/abstract/editor-adapter.abstract.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import type {EditorState} from 'prosemirror-state';
 import {Observable, Subject} from 'rxjs';
 
 @Directive()
+// TODO: Class name `TuiEditor` must have one of the following prefixes: Abstract, TuiAbstract
+// eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (nsbarsukov) [inline projects/addon-editor/abstract/editor-adapter.abstract.ts:8]: Good point! We should prepare it before v3.0

Lets mark it as deprecated.
Then create new selector
```ts
export abstract class AbstractTuiEditor extends TuiEditor {}
```

\+ write migration (or add task to `https://github.com/Tinkoff/taiga-ui/issues/1764`)

File: projects/addon-commerce/enums/currency.ts [hunk 2]
```diff
@@ -149,6 +149,7 @@ export const enum TuiCurrency {
      * @description Alphabetic code: HKD
      * @description Countries and territories: Hong Kong
      */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
     HongKong_dollar = 'HKD',
```
Comment (nsbarsukov) [inline projects/addon-commerce/enums/currency.ts:153]: lol)

Let's make it as deprecated ?
and create new one
```ts
HongKongDollar = 'HKD',
```

\+ write migration (or add task to https://github.com/Tinkoff/taiga-ui/issues/1764)

File: projects/cdk/abstract/controller.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import {OnChanges} from '@angular/core';
 import {Subject} from 'rxjs';
 
+// TODO: Class name `TuiController` must have one of the following prefixes: Abstract, TuiAbstract
+// eslint-disable-next-line @typescript-eslint/naming-convention
 export abstract class TuiController implements OnChanges {
```
Comment (nsbarsukov) [inline projects/cdk/abstract/controller.ts:6]: Lets mark it as deprecated.
Then create new selector
```ts
export abstract class AbstractTuiController extends TuiController {}
```

\+ write migration (or add task to `https://github.com/Tinkoff/taiga-ui/issues/1764`)

File: projects/kit/pipes/filter-by-input/filter-by-input.base.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ export type ArrayElement<A> = A extends ReadonlyArray<infer T>
         : T
     : never;
 
+// eslint-disable-next-line @typescript-eslint/naming-convention
 export abstract class TuiFilterByInputBase {
```
Comment (nsbarsukov) [inline projects/kit/pipes/filter-by-input/filter-by-input.base.ts:17]: the same

File: .eslintrc.js [hunk 5]
```diff
@@ -55,5 +55,48 @@ module.exports = {
...
+            },
+            {
+                selector: 'class',
+                format: ['PascalCase'],
+                // prefix: ['Tui', 'Example', 'Test', 'Mock'], // TODO: enable later
+            },
+            {
+                selector: 'function',
+                format: ['camelCase', 'PascalCase'],
+                // prefix: ['tui', 'Tui', 'mock', 'test'], // TODO: enable later
```
Comment (nsbarsukov) [inline .eslintrc.js:72]: i have doubts about prefix.
Can we configure the rule to enable prefix-restriction only for exported function/class/etc (only public API) ?

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 5 above)
Comment (splincode) [reply .eslintrc.js:72]: about this, there you can configure for non-exported

but in fact everything is exported in the project, so the maximum that I can do is to exclude the demo and scripts directories and spec.ts files

Comment (splincode) [reply .eslintrc.js]: fixed

File: projects/cdk/abstract/controller.ts (hunk 3 above)
Comment (splincode) [reply projects/cdk/abstract/controller.ts:9]: fixed, and add task to https://github.com/Tinkoff/taiga-ui/issues/1764

File: projects/addon-commerce/enums/currency.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-commerce/enums/currency.ts:159]: fixed, and add task to https://github.com/Tinkoff/taiga-ui/issues/1764

File: .eslintrc.js [hunk 6]
```diff
@@ -55,5 +55,52 @@ module.exports = {
...
+            {
+                selector: 'interface',
+                modifiers: ['exported'],
+                format: ['PascalCase'],
+                // prefix: ['Tui', 'Example', 'Test'], // TODO: enable later
+            },
+            {
+                selector: 'variable',
+                format: ['camelCase', 'UPPER_CASE', 'PascalCase', 'snake_case'],
+                leadingUnderscore: 'allowSingleOrDouble',
```
Comment (waterplea) [inline .eslintrc.js:85]: Let's get rid of underscores.

File: .eslintrc.js [hunk 7]
```diff
@@ -55,5 +55,52 @@ module.exports = {
...
+            },
+            {
+                selector: 'enum',
+                modifiers: ['exported'],
+                format: ['StrictPascalCase'],
+                // prefix: ['Tui', 'Example'], // TODO: enable later
+            },
+            {
+                selector: 'enumMember',
+                format: ['PascalCase', 'UPPER_CASE'],
```
Comment (waterplea) [inline .eslintrc.js:102]: Why `UPPER_CASE`?

**Review by waterplea — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

File: .eslintrc.js (hunk 6 above)
Comment (splincode) [reply .eslintrc.js]: fixed

File: .eslintrc.js [hunk 8]
```diff
@@ -55,5 +55,52 @@ module.exports = {
...
+            },
+            {
+                selector: 'interface',
+                modifiers: ['exported'],
+                format: ['PascalCase'],
+                // prefix: ['Tui', 'Example', 'Test'], // TODO: enable later
+            },
+            {
+                selector: 'variable',
+                format: ['camelCase', 'UPPER_CASE', 'PascalCase', 'snake_case'],
```
[the remark this answers is not in the document]
Comment (splincode) [reply .eslintrc.js]: fixed

File: .eslintrc.js (hunk 7 above)
Comment (splincode) [reply .eslintrc.js]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

File: .eslintrc.js [hunk 9]
```diff
@@ -55,5 +55,52 @@ module.exports = {
...
+            {
+                selector: 'class',
+                modifiers: ['exported'],
+                format: ['PascalCase'],
+                // prefix: ['Tui', 'Example', 'Test', 'Mock'], // TODO: enable later
+            },
+            {
+                selector: 'function',
+                modifiers: ['exported'],
+                format: ['camelCase', 'PascalCase'],
```
[the remark this answers is not in the document]
Comment (waterplea) [reply .eslintrc.js:73]: First example is a mistake which will be fixed in 3.0, second is a mixin so it kinda "pretends" to be a class. I wonder what should we do about it, @nsbarsukov @vladimirpotekhin

[the remark this answers is not in the document]
Comment (splincode) [reply .eslintrc.js]: fixed

**Pull request merged**